### PR TITLE
chore: add Claude Code skill for code review

### PR DESCRIPTION
**Pull request by nabinchha**
## 📋 Summary

Adds a new Claude Code skill (`review-code`) that enables comprehensive code reviews of the current branch or a specific GitHub PR directly from the CLI.

### ✨ Added
- [`.claude/skills/review-code/SKILL.md`](https://github.com/NVIDIA-NeMo/DataDesigner/blob/nmulepati/chore/skill-for-code-review/.claude/skills/review-code/SKILL.md) — New skill definition with:
  - Two review modes: **Branch mode** (review current branch vs base) and **PR mode** (review a PR by number)
  - Multi-pass review methodology (correctness, design, standards)
  - Integration with project guidelines from `AGENTS.md`
  - Structured output format with severity-based findings (Critical, Warnings, Suggestions)
  - Linter integration via `ruff`

---
🤖 *Generated with AI*

### CODE REVIEW

File: .claude/skills/review-code/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
+---
+name: review-code
+description: Perform a thorough code review of the current branch or a GitHub PR by number. Use when the user asks to review code changes, audit a branch or PR, or wants a pre-merge quality check.
```
Comment (johnnygreco) [inline .claude/skills/review-code/SKILL.md]: The last bet about "Use when..." is only needed with model invocation, right? Of course, having it work when you ask for a review is an option. Just noting that it is a bit inconsistent with the setting below.

File: .claude/skills/review-code/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+- Are edge cases tested?
+- Are mocks/stubs used appropriately (at boundaries, not deep internals)?
+- Do new test names clearly describe what they verify?
+
+**Project Standards (from AGENTS.md) — apply to new/modified code only:**
+
+Verify the items below on lines introduced or changed by this branch. Refer to the full `AGENTS.md` loaded in Step 2 for details and examples.
+
+- License headers (`SPDX-FileCopyrightText` / `SPDX-License-Identifier`) on new files
+- `from __future__ import annotations` in new files
```
Comment (johnnygreco) [inline .claude/skills/review-code/SKILL.md]: Do we want  it to worry about things that are (supposed to be) caught by our CI?

File: .claude/skills/review-code/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+```bash
+uv run ruff check <changed-files>
+uv run ruff format --check <changed-files>
+```
+
+If the branch isn't checked out locally (e.g., external fork in PR mode), skip this step and note it in the review.
+
+## Step 6: Produce the Review
+
+Output a structured review using the format below. Use the **PR template** or **Branch template** for the overview depending on the mode.
```
Comment (johnnygreco) [inline .claude/skills/review-code/SKILL.md:195]: Should we ask for a markdown file specifically here?

File: .claude/skills/review-code/SKILL.md (hunk 2 above)
Comment (andreatnvidia) [reply .claude/skills/review-code/SKILL.md]: good point. the `/code-review` skill that ships with claude code has an explicit false positive list that could be a good reference here. it tells the reviewer to skip things like linter/typechecker catches, pre-existing issues on unmodified lines, pedantic nits, and intentional functionality changes. could add something similar to keep the review focused on stuff CI won't catch for us.

File: .claude/skills/review-code/SKILL.md (hunk 3 above)
Comment (andreatnvidia) [reply .claude/skills/review-code/SKILL.md:195]: +100. make them temporary though, shouldn't commit. but storing them for a while should allow us to use other agents to handle the output.

File: .claude/skills/review-code/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+1. **PR details**: `gh pr view <number> --json title,body,author,baseRefName,headRefName,state,additions,deletions,changedFiles,commits,url`
+2. **PR diff**: `gh pr diff <number>`
+3. **PR files**: `gh pr diff <number> --name-only`
+4. **PR commits**: `gh pr view <number> --json commits --jq '.commits[].messageHeadline'`
+5. **Existing review comments**: `gh api repos/{owner}/{repo}/pulls/<number>/comments --jq '.[].body'`
+6. **Repo info**: `gh repo view --json nameWithOwner -q '.nameWithOwner'`
+
+Then checkout the PR branch for full file access:
+
+```bash
```
Comment (andreatnvidia) [inline .claude/skills/review-code/SKILL.md:45]: `gh pr checkout` switches your current branch, which can mess things up if you have uncommitted work. a worktree would be safer:

```bash
git fetch origin pull/<number>/head:pr-<number>
git worktree add checkouts/review-<number> pr-<number>
```

reviews in isolation, then just `git worktree remove` when you're done.

File: .claude/skills/review-code/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+### If PR mode (argument starts with a number)
+
+Run these commands in parallel using `gh`:
+
+1. **PR details**: `gh pr view <number> --json title,body,author,baseRefName,headRefName,state,additions,deletions,changedFiles,commits,url`
+2. **PR diff**: `gh pr diff <number>`
+3. **PR files**: `gh pr diff <number> --name-only`
+4. **PR commits**: `gh pr view <number> --json commits --jq '.commits[].messageHeadline'`
+5. **Existing review comments**: `gh api repos/{owner}/{repo}/pulls/<number>/comments --jq '.[].body'`
```
Comment (andreatnvidia) [inline .claude/skills/review-code/SKILL.md]: +1 to the greptile comment. you could also do this with a single graphql query that grabs both review threads and top-level review bodies at once:

```graphql
query($owner: String!, $repo: String!, $pr: Int!) {
  repository(owner: $owner, name: $repo) {
    pullRequest(number: $pr) {
      reviews(first: 50) {
        nodes { author { login } state body }
      }
      reviewThreads(first: 100) {
        nodes {
          isResolved
          comments(first: 10) {
            nodes { author { login } body path line }
          }
        }
      }
    }
  }
}
```

one round trip, and you also get resolved vs unresolved for free.

File: .claude/skills/review-code/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+- Lazy loading for heavy third-party imports via `lazy_heavy_imports` + `TYPE_CHECKING`
+- Naming: snake_case functions starting with a verb, PascalCase classes, UPPER_SNAKE_CASE constants
+- No vacuous comments — comments only for non-obvious intent
+- Public before private ordering in new classes
+- Design principles: DRY (extract on third occurrence), KISS (flat over clever), YAGNI (no speculative abstractions)
+- Common pitfalls: no mutable default arguments, no unused imports, simplify where possible
+
+## Step 5: Run Linter
+
+Run the linter on all changed files (requires local checkout):
```
Comment (andreatnvidia) [inline .claude/skills/review-code/SKILL.md]: nit: `uv run` hits sandbox permission issues in claude code (it tries to write to `~/.cache/uv`). running from `.venv/bin/` directly works fine:

```bash
.venv/bin/ruff check <changed-files>
.venv/bin/ruff format --check <changed-files>
```

File: .claude/skills/review-code/SKILL.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md]: Good call, added in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95.

File: .claude/skills/review-code/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+```bash
+uv run ruff check <changed-files>
+uv run ruff format --check <changed-files>
+```
```
[the remark this answers is not in the document]
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md:187]: Updated in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95

File: .claude/skills/review-code/SKILL.md (hunk 2 above)
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md]: Good call out, updated instructions in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95

File: .claude/skills/review-code/SKILL.md (hunk 3 above)
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md:195]: I updated instructions in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95. When I was testing I just continued the conversation to dive deeper, but makes sense to persist this into a temp md file.

File: .claude/skills/review-code/SKILL.md (hunk 4 above)
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md:45]: Good call. Updated in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95.

File: .claude/skills/review-code/SKILL.md (hunk 5 above)
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md]: Claude wants to keep it rest based with 2 round trips for now :) d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95

File: .claude/skills/review-code/SKILL.md (hunk 6 above)
Comment (nabinchha) [reply .claude/skills/review-code/SKILL.md]: updated in d0d2361b8d1a3b86ab268cc445a7df3a61f1dc95

**Review by andreatgretel — APPROVED** [review]
